meowlflow/api/middlewares/errors.py

```diff
@@ -9,25 +9,48 @@
 logger = logging.getLogger(__name__)
 
 
-async def catch_exceptions_middleware(
-    request: Request, call_next: Callable[[Request], Awaitable[Response]]
-) -> Response:
-    try:
-        return await call_next(request)
-    except MeowlflowException as error:
-        # you probably want some kind of logging here
-        logger.error(error)
-        logger.error(traceback.format_exc())
-        return JSONResponse(
-            {"error": error.to_dict()},
-            status_code=error.status_code,
-        )
-
-    except Exception as err:  # pylint: disable=broad-except
-        logger.error(err)
-        logger.error(traceback.format_exc())
-        err = MeowlflowException("Internal server error", {})
-        return JSONResponse(
-            {"error": err.to_dict()},
-            status_code=err.status_code,
-        )
+def configure_catch_exceptions_middleware(handlers=[]):
+    async def catch_exceptions_middleware(
+        request: Request, call_next: Callable[[Request], Awaitable[Response]]
+    ) -> Response:
+        try:
+            return await call_next(request)
+
+        except MeowlflowException as error:
+            logger.error(error)
+            logger.error(traceback.format_exc())
+
+            for handler in handlers:
+                try:
+                    handler(error)
+                except Exception as handler_error:
+                    logger.error(handler_error)
+                    logger.error(traceback.format_exc())
+
+            return JSONResponse(
+                {"error": error.to_dict()},
+                status_code=error.status_code,
+            )
+
+        # now try to catch all unexpected Exceptions
+        except Exception as error:  # pylint: disable=broad-except
```

Comment:
the downside of capturing errors this way is that our integrations (like sentry)  believe them to be "handled"
![image](https://user-images.githubusercontent.com/11619412/214041435-b8baaf9c-3012-4556-9dbd-78724e2fda05.png)

Reply:
ALSO!
the current `sentry.handler` causes all local variables to be lost, and all of the good automatic sentry tracing is not included.

Reply:
Wait, that seems like a bad tradeoff, if we don't get any of the variables or tracing!